submissions/kyldvs/src/SithDataManager.js

```diff
@@ -0,0 +1,64 @@
+/**
+ * @flow
+ */
+
+'use strict';
+
+import Dispatcher from './Dispatcher';
+import Sith from './Sith';
+import xhr from 'xhr';
+
+const URL = 'http://localhost:3000/dark-jedis/';
+
+// Keep track of requests so we can abort them later.
+let counter = 1;
+const requests = new Map();
```

Comment:
Ahh great point! I wasn't thinking so much about trimming that aspect of it down. I think a map is still the way to go for a few reasons.

1. This makes it easy to support changes for when requests can be parallelized. In fact I do have one case in this example where this is required. When you start loading Siths if you put DarthSidius as the first Sith then there are only enough apprentices to fill the first 3 indices in the list. So I actually have him starting at the center spot. This means initially two requests can be in flight, one for the apprentice, and one for the master.

2. This doesn't enable me to get rid of `requestID` since I still need it to map back the response of a request to the appropriate slot in the SithStore.

Good thought though. I think I will stick with map because I believe it is a less error prone and a more future proof way to implement it.